src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -98,6 +127,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _defaultAppearanceViewModel.IsDefault(true);
     }
 
+    void ProfileViewModel::_UpdateBuiltInIcons()
+    {
+        _BuiltInIcons = single_threaded_vector<IInspectable>();
+        for (auto& [val, name] : s_SegoeFluentIcons)
+        {
+            _BuiltInIcons.Append(make<EnumEntry>(hstring{ name }, box_value(val)));
+        }

```

Comment:
I'd recommend the "first construct a `std::vector`, then move it into an `IVector`" approach here given that you append a lot of items.

BTW it's a funny thought but technically it's completely 100% possible to create a custom HSTRING that does not the need a heap allocation. The HSTRING struct is part of the ABI.

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/winstring/nf-winstring-windowscreatestringreference eh